### ISSUE TITLE
active-gtk-theme related refs are not installed when using FlatpakTransaction

### ISSUE THREAD

**Issue by AsciiWolf** [flatpak/flatpak#1920]
## Linux distribution and version
Fedora Rawhide

## Flatpak version
flatpak 0.99.3
gnome-software 3.29.2 (latest master)

## Description of the problem
Also reported [here](https://gitlab.gnome.org/GNOME/gnome-software/issues/449), but I am not sure if this is really a GS issue so I am reporting it as a Flatpak issue as well.

If I try to install for example Telegram (or VLC) using the latest GNOME Software that uses the new FlatpakTransaction API and try to run `flatpak update` afterwards, one additional related ref is being installed although it should be installed during Telegram/VLC installation:
`$ flatpak update`
`Looking for updates...`
`Installing in system:`
`org.kde.KStyle.Adwaita/x86_64/5.9  flathub_1 41f06d286c71`
`Is this ok [y/n]: `

If I install Telegram manually using `flatpak install flathub_1 org.telegram.desktop`, all related refs are properly installed.

## Steps to reproduce
1. Use latest, fully updated Fedora Rawhide.
2. Add Flathub repo:
`flatpak remote-add --if-not-exists flathub https://flathub.org/repo/flathub.flatpakrepo`
3. Download, compile and run latest GNOME Software:
`wget https://gitlab.gnome.org/GNOME/gnome-software/-/archive/master/gnome-software-master.tar.bz2`
`tar xvf gnome-software-master.tar.bz2 && cd gnome-software-master`
`meson --prefix $PWD/install build/`
`ninja -C build/ all install`
`killall gnome-software`
`XDG_DATA_DIRS=install/share:$XDG_DATA_DIRS ./install/bin/gnome-software`
4. Find and install _Telegram_ or _VLC_.
5. Manually run `flatpak update` after the installation.

**Comment by AsciiWolf** [comment]
I have just tried installing more than 20 Flatpaks using GS from master and there were no issues with other related refs except the _org.kde.KStyle.Adwaita_ one. So it looks like that _org.kde.KStyle.Adwaita_ is the only one that FlatpakTransaction API (or GS) has problems with.

**Comment by alexlarsson** [comment]
That makes it sounds like it only affects extension points with `download-if=active-gtk-theme`.

**Comment by alexlarsson** [comment]
Is XDG_DATA_DIRS or something like that set to any weird values?

**Comment by AsciiWolf** [comment]
@alexlarsson I am currently not at home till next week so I am not able to check this, but I had the same issue on both Fedora Rawhide and Ubuntu 18.10 with latest GNOME Software 3.30 (3.29) build. You can try asking @hughsie.

**Comment by AsciiWolf** [comment]
On the Fedora Rawhide machine:
```
$ env|grep XDG_DATA_DIRS
XDG_DATA_DIRS=/home/asciiwolf/.local/share/flatpak/exports/share/:/var/lib/flatpak/exports/share/:/usr/local/share/:/usr/share/
```

**Comment by alexlarsson** [comment]
@AsciiWolf So, it seem to me like the issue is with `download-if=active-gtk-theme` extensions, and those get downloaded by the CLI by this code:

```
  else if (strcmp (reason, "active-gtk-theme") == 0)
    {
      const char *gtk_theme = flatpak_get_gtk_theme ();
      return strcmp (gtk_theme, extension_basename) == 0;
    }
```

Can you put a breakpoint or something on flatpak_get_gtk_theme  in both gnome-software and the CLI and see if it somehow returns different values in the two?

**Comment by AsciiWolf** [comment]
@alexlarsson Do you have any tip how to debug this? I have tried gdb and strace, but didn't get any detailed info. Here is at least some strace output of installing org.telegram.desktop with "gtk" string grepped: [Flatpak](https://pastebin.com/V63Vvj0f), [GNOME Software](https://pastebin.com/M69t1DBD)

Can someone else try this too (using the provided reproducer)?

**Comment by alexlarsson** [comment]
Something like:
``'
gdb attach -p $(pidof gnome-software)
(gdb) break flatpak_get_gtk_theme
(gdb) continue
.. do install
(gdb) bt
```

**Comment by AsciiWolf** [comment]
As I mentioned earlier on IRC, I am not sure if this is a Flatpak issue or a GS one, but it would be great if this could be fixed before the GNOME 3.30 (and Fedora 29) release as it makes all Qt Flatpak apps installed using GS look bad (and GTK apps look default when using custom GTK theme).

**Comment by alexlarsson** [comment]
cool
